src/System.Drawing.Design.Primitives/src/System/Drawing/Design/ToolboxItem.cs

```diff
@@ -0,0 +1,1100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Runtime.Serialization;
+using System.ComponentModel;
+using System.ComponentModel.Design;
+using System.Diagnostics;
+using System.Reflection;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Permissions;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.Versioning;
+
+namespace System.Drawing.Design
+{
+    /// <summary>
+    /// Provides a base implementation of a toolbox item.
+    /// </summary>
+    [Serializable]
+    public class ToolboxItem : ISerializable
+    {
+        private static TraceSwitch s_toolboxItemPersist = new TraceSwitch("ToolboxPersisting", "ToolboxItem: write data");
+
+        private static object s_eventComponentsCreated = new object();
+        private static object s_eventComponentsCreating = new object();
+
+        private bool _locked;
+        private LockableDictionary _properties;
+        private ToolboxComponentsCreatedEventHandler _componentsCreatedEvent;
+        private ToolboxComponentsCreatingEventHandler _componentsCreatingEvent;
+
+        /// <summary>
+        /// Initializes a new instance of the ToolboxItem class.
+        /// </summary>
+        public ToolboxItem() { }
+
+        /// <summary>
+        /// Initializes a new instance of the ToolboxItem class using the specified type.
+        /// </summary>
+        /// <param name="toolType"></param>
+        public ToolboxItem(Type toolType) : this()
+        {
+            Initialize(toolType);
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref='System.Drawing.Design.ToolboxItem'/>
+        ///class using the specified serialization information.
+        /// </summary>
+        protected ToolboxItem(SerializationInfo info, StreamingContext context) : this()
+        {
+            Deserialize(info, context);
+        }
+
+        /// <summary>
+        /// The assembly name for this toolbox item. The assembly name describes the assembly
+        ///information needed to load the toolbox item's type.
+        /// </summary>
+        public AssemblyName AssemblyName
+        {
+            get
+            {
+                return (AssemblyName)Properties["AssemblyName"];
+            }
+            set
+            {
+                Properties["AssemblyName"] = value;
+            }
+        }
+
+        /// <summary>
+        /// The assembly name for this toolbox item. The assembly name describes the assembly
+        /// information needed to load the toolbox item's type.
+        /// </summary>
+        public AssemblyName[] DependentAssemblies
+        {
+            get
+            {
+                AssemblyName[] names = (AssemblyName[])Properties["DependentAssemblies"];
```

Comment:
Nit: if we wanted to, we could use `nameof(DependentAssemblies)` instead of `"DependentAssemblies"` (same for all the other such properties) to avoid any issues related to typos in hand-crafted strings.